scipy/spatial/distance.py

```diff
@@ -144,6 +146,67 @@ def _validate_vector(u, dtype=None):
         raise ValueError("Input vector should be 1-D.")
     return u
 
+def directed_hausdorff(u, v):
+    """
+    Computes the directed Hausdorff distance between N-dimensional
+    point sets using a Euclidean metric.
+    
+    .. versionadded:: 0.19.0
+
+    Parameters
+    ----------
+    u : (M,N) ndarray
+        Input array.
+    v : (O,N) ndarray
+        Input array.
+
+    Returns
+    -------
+    d : double
+        The directed Hausdorff distance between arrays `u` and `v`.
+
+    Notes
+    ----------
+    Uses the early break technique and the random sampling approach described
+    by [Taha2015]_. Although worst-case performance is polynomial (as with the
```

Comment:
Would be good to be more exact than "polynomial". From a quick read of the paper, it's ``O(m * o)``.